enhancements/dns/externaldns-operator.md

```diff
@@ -0,0 +1,467 @@
+Note the optional `providerSpecific` field, in which name/value pairs can be added as necessary.
+
+Note that currently, the ExternalDNS CRD source implementation only updates the `observedGeneration` status subfield.
+An [issue has been opened upstream](https://github.com/kubernetes-sigs/external-dns/issues/2092) to better understand
+upstream's intentions with CRD source status fields. ExternalDNS does provide adequate debug logs, should a record fail to update,
+in addition to a few [high-level metrics](https://github.com/kubernetes-sigs/external-dns/blob/master/controller/controller.go).
+However, status fields to reflect the availability of the requested record in the given provider would be useful for other controllers running
+in a cluster, as well as for cluster administrators who want to verify that their requested resources have been created properly.
+
+Currently, the upstream ExternalDNS CRD source only supports record creation for A or CNAME type records. See
+[upstream issue 164](https://github.com/kubernetes-sigs/external-dns/issues/164) for more context to that end. Contributions may need to made to
+the upstream CRD source implementation to allow for other record types to be created via the DNSRecord CRD on OpenShift, such as `TXT` or `AAAA` records.
+
+__Note:__ Any API(s) introduced by this enhancement will follow the Kubernetes API
+[deprecation policy](https://kubernetes.io/docs/reference/using-api/deprecation-policy/). Example API groups are also subject to change.
+
+### User Stories [optional]
+
+#### Story 1
+
+As a developer, I need the ability to manage DNS records for OpenShift routes without manual creating
+and updating DNS records by hand.
+
+#### Story 2
+
+As a cluster administrator, I need the ability use a single operator to manage external DNS requirements for all
+platform components.
+
+#### Story 3
+
+As a security-mindful cluster administrator, I cannot in good conscious let the Ingress Operator create wildcard DNS records
+for cluster Ingress. I need to have the ability to manage fully-qualified DNS records for cluster workloads with Routes.
+
+#### Story 4
+
+As a cluster administrator, I want ExternalDNS to create DNS records for all of my LoadBalancer services in Route53,
+in addition to creating DNS records for all of my OpenShift routes in BlueCat DNS, so that I can easily manage my multi-provider
+DNS requirements.
+
+To satisfy this, a cluster administrator could create 2 different ExternalDNS Operator Custom Resources, one for each cloud provider.
+
+#### Story 5
+
+As a cluster administrator, I want the ExternalDNS Operator to deploy one ExternalDNS instances for each of my DNS zones, with each deployment
+being configured to use a different set of credentials, since I do not want to reuse common credentials for both zones due to security and auditing concerns.
+
+### Implementation Details [optional]
```

Comment:
`[optional]` can be removed since the details are included in the EP.

Reply:
Right! Fixed.